### PR TITLE
add attr scale. test=develop

### LINKED ISSUE #16362: resize_nearest的scale参数不是动态scale
resize_nearest的scale参数不是动态scale，而是在构图时转换为out_h和out_w，这样对于动态shape输入会有影响

### LINKED ISSUE #16384: resize_bilinear层没有保存scale参数
paddle.fluid.layers.resize_bilinear层，在保存模型的时候没有保存scale参数。无法支持变长的模型输入。
paddle版本1.3.0

### PR DESCRIPTION
**Pull request by heavengate**
fix https://github.com/PaddlePaddle/Paddle/issues/16362
fix https://github.com/PaddlePaddle/Paddle/issues/16384

红框中为文档修改内容
MacHi 2019-04-01 09-17-51

### CODE REVIEW

File: python/paddle/fluid/layers/nn.py [hunk 1]
```diff
@@ -7056,10 +7056,10 @@ def image_resize(input,
         out_shape(list|tuple|Variable|None): Output shape of image resize
                                     layer, the shape is (out_h, out_w).
                                     Default: None
-        scale(float|None): The multiplier for the input height or width.
-                         At least one of out_shape or scale must be set.
-                         And out_shape has a higher priority than scale.
-                         Default: None
+        scale(float|None): The multiplier for the input height or width. At
+             least one of :attr:`out_shape` or :attr:`scale`` must be set. 
+             And :attr:`scale` has a higher priority than :attr:`out_shape`.
```
Comment (qingqing01) [inline python/paddle/fluid/layers/nn.py]: Why `scale` has higher priority? 

Maybe only when `out_shape` is a Variable, need to set `scale`  for the infer-shape?

Comment (heavengate) [reply python/paddle/fluid/layers/nn.py]: doc here has mistake， out_shape has higher priority than scale in current implement, which is compatible with old version

**Review by qingqing01 — DISMISSED** [review]
LGTM.

**Comment by qingqing01** [comment]
@NHZlX please double check.

File: paddle/fluid/operators/interpolate_op.cc [hunk 2]
```diff
@@ -36,10 +36,18 @@ class InterpolateOp : public framework::OperatorWithKernel {
...
     auto dim_x = ctx->GetInputDim("X");  // NCHW format
-    int out_h = ctx->Attrs().Get<int>("out_h");
-    int out_w = ctx->Attrs().Get<int>("out_w");
     PADDLE_ENFORCE_EQ(dim_x.size(), 4, "X's dimension must be 4");
 
+    int out_h, out_w;
+    float scale = ctx->Attrs().Get<float>("scale");
+    if (scale > 0) {
+      out_h = dim_x[2] * scale;
+      out_w = dim_x[3] * scale;
```
Comment (qingqing01) [inline paddle/fluid/operators/interpolate_op.cc]: Maybe add comments:  there is rounding here.

File: paddle/fluid/operators/interpolate_op.cc [hunk 3]
```diff
@@ -36,10 +36,18 @@ class InterpolateOp : public framework::OperatorWithKernel {
         "Interpolation method can only be \"bilinear\" or \"nearest\".");
 
     auto dim_x = ctx->GetInputDim("X");  // NCHW format
-    int out_h = ctx->Attrs().Get<int>("out_h");
-    int out_w = ctx->Attrs().Get<int>("out_w");
     PADDLE_ENFORCE_EQ(dim_x.size(), 4, "X's dimension must be 4");
 
+    int out_h, out_w;
+    float scale = ctx->Attrs().Get<float>("scale");
```
Comment (NHZlX) [inline paddle/fluid/operators/interpolate_op.cc:42]: 对于之前训练的模型是没有这个attr的， ```float scale = ctx->Attrs().Get<float>("scale");``` 之前的模型运行到这行代码的时候应该会报错吧...

Comment (qingqing01) [reply paddle/fluid/operators/interpolate_op.cc:42]: 这个attr给了默认值，这个check下不使用attr时保存的ProgramDesc，看看能否加载？

Comment (heavengate) [reply paddle/fluid/operators/interpolate_op.cc:42]: 测试了下，用老版的input+resize_bilinear的ProgramDesc里attr没有scale，并save_inference_model, 用带Attr(scale)版的ProgramDesc中有attr有scale， load_inference_model后能正确跑infer

**Review by qingqing01 — DISMISSED** [review]

**Comment by heavengate** [comment]
@shanyi15 plz check doc

**Comment by qingqing01** [comment]
Please check the doc, whether the `out_shape` is Variable.

**Comment by heavengate** [comment]
> Please check the doc, whether the `out_shape` is Variable.

Done

**Review by shanyi15 — DISMISSED** [review]
LG comments

**Review by qingqing01 — DISMISSED** [review]

**Review by shanyi15 — DISMISSED** [review]

**Review by qingqing01 — APPROVED** [review]

**Review by shanyi15 — APPROVED** [review]

**Review by XiaoguangHu01 — APPROVED** [review]
LGTM

**Pull request merged**